tests/foreman/ui/test_architecture.py

```diff
@@ -32,14 +34,17 @@ def test_positive_create_arch_with_os(self, test_data):
         @Assert: Architecture is created
 
         """
-        entities.OperatingSystem(name=test_data['os_name']).create()
         with Session(self.browser) as session:
-            make_arch(session, name=test_data['name'],
-                      os_names=[test_data['os_name']])
-            self.assertIsNotNone(self.architecture.search(test_data['name']))
-
-    @data(*generate_strings_list())
-    def test_positive_create_arch_with_different_names(self, name):
+            for test_data in valid_arch_os_names():
+                with self.subTest(test_data):
+                    entities.OperatingSystem(
+                        name=test_data['os_name']).create()
+                    make_arch(session, name=test_data['name'],
```

Comment:
@oshtaier see https://github.com/SatelliteQE/robottelo/issues/2909.  Fixing #2909 is beyond the scope of this PR because it involves a framework fix + change all locator references + retest.  When this issue is addressed, html values can be put back here.